apps/aecontract/src/aect_contracts.erl

```diff
@@ -67,8 +68,8 @@
         id           :: aeser_id:id(),
         owner_id     :: aeser_id:id(),
         ct_version   :: version(),
-        code         :: binary(),     %% The byte code
-        code_ref     :: no_ref | {ref, aeser_id:id()},
+        code         :: binary(),     %% The byte code when code_ref=no_ref
+        code_ref     :: no_ref | {ref, aeser_id:id()},     %% no_ref or DIRECT reference to contract with bytecode
```

Comment:
Why not 
```suggestion
        code     :: {code, binary()} | {ref, aeser_id:id()},
```

Reply:
To be honest... no reason. I will change that. This is kind of internal working of aect_contracts. And it will be better if I combine the fields. First I did it like this to reflect serialized and deserialized form where we don't have access to complex types.

Reply:
Done